src/server/scripts/Outland/zone_blades_edge_mountains.cpp

```diff
@@ -30,78 +30,178 @@ EndContentData */
+    NPC_EXPLOSION_BUNNY                         = 22502,
+    NPC_FEL_IMP                                 = 22474,
+    NPC_HOUND                                   = 22500,
     NPC_NORTH_GATE                              = 22471,
     NPC_SOUTH_GATE                              = 22472,
     NPC_NORTH_GATE_CREDIT                       = 22503,
     NPC_SOUTH_GATE_CREDIT                       = 22504,
 
+    GO_FIRE                                     = 185317,
+    GO_BIG_FIRE                                 = 185319,
+
+    EVENT_PARTY_TIMER                           = 1
 };
 
-class npc_deahts_door_wrap_gate : public CreatureScript
+class npc_deaths_door_fell_cannon_target_bunny : public CreatureScript
 {
 public:
-    npc_deahts_door_wrap_gate() : CreatureScript("npc_deahts_door_wrap_gate") { }
+    npc_deaths_door_fell_cannon_target_bunny() : CreatureScript("npc_deaths_door_fell_cannon_target_bunny") { }
 
-    CreatureAI* GetAI(Creature* creature) const
+    struct npc_deaths_door_fell_cannon_target_bunnyAI : public ScriptedAI
     {
-        return new npc_deahts_door_wrap_gateAI(creature);
-    }
+        npc_deaths_door_fell_cannon_target_bunnyAI(Creature* creature) : ScriptedAI(creature), PartyTime(false) { }
 
-    struct npc_deahts_door_wrap_gateAI : public ScriptedAI
-    {
-        npc_deahts_door_wrap_gateAI(Creature *c) : ScriptedAI(c)
+        EventMap events;
+        bool PartyTime;
+        uint64 PlayerGUID;
+        uint64 CannonGUID;
+        uint8 count;
+        
+        void Reset() override
         {
+            Initialize();
+            events.Reset();
         }
 
-        uint8 count;
-        uint32 timer;
-        void Reset() { count = 0; timer = 0;}
-        void SpellHit(Unit* caster, SpellInfo const* spellInfo)
+        void Initialize()
         {
-            if (spellInfo->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
-            {
-                Player* plr = caster->GetCharmerOrOwnerPlayerOrPlayerItself();
-                if (!plr)
-                    return;
+            PartyTime = false;
+            PlayerGUID = 0;
+            CannonGUID = 0;
+            count = 0;
+        }
 
-                timer = 1;
+        void SpellHit(Unit* caster, SpellInfo const* spell) override
+        {
+            if (spell->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
+            {
                 count++;
 
+                if (count)
+                {
+                    if (Player* player = caster->GetCharmerOrOwnerPlayerOrPlayerItself())
+                        PlayerGUID = player->GetGUID();
+
+                    CannonGUID = caster->GetGUID();
+                    PartyTime = true;
+                    events.ScheduleEvent(EVENT_PARTY_TIMER, 3000);
+                }
+
+                if (count >= 3)
+                    me->SummonGameObject(GO_FIRE, me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 130);
+
                 if (count >= 6)
```

Comment:
It should take 7 shots to take the warp gate down.
```suggestion
                if (count > 6)
```